templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,9 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       var selected_status_id  = obj.find(":selected").val();
+       var cancelInfo_show_ids = [{/literal}{$cancelInfo_show_ids}{literal}];
+       if (cancelInfo_show_ids.indexOf(selected_status_id) > -1) {
```

Comment:
> hmm, it looks like you are searching a string and not an array.

No, the value of the smarty variable ``cancelInfo_show_ids`` is 
``
'8','12'
``
which turns line 492 into
``
 var cancelInfo_show_ids = ['8','12'];
``
So it works on an array of strings, which seems correct to me - option group values could be strings, after all. 
I preferred this approach over the ``json_encode`` based one, because it still produces valid JS if the Smarty variable is not set (for whatever reason).